tsc/frr/sonic_frr_update_process.md

```diff
@@ -0,0 +1,79 @@
+# **SONiC Community FRR Upgrade Process Proposal**
+
+FRR upgrade and patch cadence is largely on need basis in the current SONiC release program. This SONiC FRR upgrade process proposal is to formalize the FRR upgrade cadence and process for future SONiC release programs.  
+
+# SONiC FRR Version Upgrade History
+
+<p align=center>
+<img src="frr.png" alt="">
+</p>
+
+
+# FRR Project Release Cadence
+-   FRR release numbering scheme x.y.z-s#
+-   New FRR releases roughly every 4 months
+	- Apr 2023 - Release 8.5.1 (48 fixes)
+	- Mar 2023 - Release 8.5 (947+ commits)
+	- Jan 2023 - Release 8.4.2 (22 fixes)
+	- Nov 2022 - Release 8.4.1 (16 fixes)
+	- Nov 2022 - Release 8.4 (700+ commits)
+	- Aug 2022 - Release 8.3.1 (14 fixes)
+	- Jul 2022 - Release 8.3 (1000+ commits)
+	- Mar 2022 - Release 8.2.2 (800+ commits)
+	- Nov 2021 - Release 8.1.0 (1200+ commits)
+	- Jul 2021 - Release 8.0.0 (2200+ commits)
+
+-   SONiC to stay out from major/minor releases (x.y) and use patch release (.z) for stability (eg, FRR 8.3.1 instead of 8.3 if it is for 202211 release)
+
+# SONiC Release FRR Upgrade
+-   SONiC default to rebase FRR in every November community release
+-   SONiC FRR upgrade test requirements
+	-   MANDATORY: Pass all Azure pipeline build test and LGTM as required by the standard code PR merge process 
+  	-   OPTIONAL: Additional tests in respect to specific changeset in the upgrade as deem necessary, manual tests should be automated and submitted to improve future test coverage   
+-   Rotate SONiC FRR maintenance duty among repo maintainer org and others (BRCM, MSFT, Alibaba, NVDA, DELL)
+-   Responsibility of SONiC FRR release maintainer
+	-   Default 12 months assignment
+	-   Upgrade FRR version in Nov release, resolve SONiC FRR upgrade integration issues
+	-   Triage and fix SONiC FRR issues when applicable. Fix may come from SONiC contributors or from FRR community, maintainer is responsible to drive the fix to unblock SONiC community
+	-   Submit fixes to FRR project, submit new FRR topo test to FRR project if there is a gap
+	-   Release maintainer to subscribe to FRR project, and be the FRR Point-of-Contact on behalf of SONiC
+	-   Bring in FRR vulnerabilities and critical patches to SONiC
+  -   If there is a need for May release upgrade due to feature requirement or other reasons, it should be requested in the May release plan and get a consensus with the FRR release maintainer. The person or organization requested the upgrade will be responsible of carrying out FRR upgrade with the same FRR upgrade process described in this proposal. FRR release maintainer will need to work closely with the person or organization to oversee the mid term upgrade, he/she will continue to assume the role of FRR release maintainer until the end of his/her term
```

Comment:
>> he/she will continue to assume the role of FRR release maintainer until the end of his/her term
until the end of regular FRR rebase in November? I believe, irrespective of FRR rebase done as part of the May release or not on a need basis, November SONiC community release will look at the changes in the FRR and rebase as appropriate.

Reply:
Correct, he/she will continue the role as a release maintainer for FRR until the full 1 year term ends in Nov. Even with May upgrade, the Nov upgrade will be assumed